scripts/download_db.sh

```diff
@@ -1,26 +1,52 @@
-# This script is used to download a nightly Berkeley database dump used for debugging purposes.
 #!/usr/bin/env bash
-set -x
-set -eo pipefail
+set -eo pipefail  
 
-PG_DUMP="berkeley-archive.sql"
+# default to 'mainnet'
+NETWORK="${1:-mainnet}"
+
+# Validate the network input
+if [[ "$NETWORK" != "mainnet" && "$NETWORK" != "devnet" ]]; then
+  echo "Usage: $0 [mainnet|devnet]"
+  exit 1
+fi
+
+# Define output paths and base URL
 DATA_DIR="data"
-CURRENT_DATE=$(date '+%Y-%m-%d')
+PG_DUMP="${NETWORK}-archive.sql"
+BASE_URL="https://storage.googleapis.com/mina-archive-dumps"
 
-# Create the data directory if it doesn't exist
-mkdir -p ${DATA_DIR}
+# Create the data directory and change into it
+mkdir -p "$DATA_DIR"
+cd "$DATA_DIR"
 
-# Change to the data directory
-cd ${DATA_DIR}
+# get date as YYYY-MM-DD
+get_date() {
+  # macOS
+  if [[ "$OSTYPE" == "darwin"* ]]; then
+    date -v -"$1"d '+%Y-%m-%d'    
+  # linux
+  else
+    date -d "-$1 days" '+%Y-%m-%d'
```

Comment:
The mac thread works on my machine TM.  Trusting that the linux branch also works.